src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,474 @@
+        )
+
+        private val enabledPattern by repoGroup.pattern(
+            "perk.enable", "§a§lENABLED|(§.)*SELECTED"
+        )
+        private val disabledPattern by repoGroup.pattern(
+            "perk.disabled", "§c§lDISABLED|§7§eClick to select!"
+        ) // unused for now since the assumption is when enabled isn't found it is disabled, but the value might be useful in the future or for debugging
+
+        private val resetChatPattern by repoGroup.pattern(
+            "reset.chat", "§aReset your §r§5Heart of the Mountain§r§a! Your Perks and Abilities have been reset."
+        )
+
+        private val heartItemPattern by repoGroup.pattern(
+            "inventory.heart", "§5Heart of the Mountain"
+        )
+        private val resetItemPattern by repoGroup.pattern(
+            "inventory.reset", "§cReset Heart of the Mountain"
+        )
+
+        private val heartTokensPattern by repoGroup.pattern(
+            "inventory.heart.token", "§7Token of the Mountain: §5(?<token>\\d+)"
+        )
+
+        private val resetTokensPattern by repoGroup.pattern(
+            "inventory.reset.token", "\\s+§8- §5(?<token>\\d+) Token of the Mountain"
+        )
+
+        var inInventory = false
+
+        var tokens: Int
+            get() = ProfileStorageData.profileSpecific?.mining?.tokens ?: 0
+            private set(value) {
+                ProfileStorageData.profileSpecific?.mining?.tokens = value
+            }
+
+        var availableTokens: Int
+            get() = ProfileStorageData.profileSpecific?.mining?.availableTokens ?: 0
+            private set(value) {
+                ProfileStorageData.profileSpecific?.mining?.availableTokens = value
+            }
+
+        var heartItem: Slot? = null
+
+        init {
+            entries.forEach { it.guiNamePattern }
+            HotmAPI.Powder.entries.forEach {
+                it.heartPattern
+                it.resetPattern
+            }
+        }
+
+        private fun resetTree() = entries.forEach {
+            it.activeLevel = 0
+            it.enabled = false
+            it.isUnlocked = false
+            HotmAPI.Powder.entries.forEach { it.setCurrent(it.getTotal()) }
+            availableTokens = tokens
+        }
+
+        private fun Slot.parse() {
+            val item = this.stack ?: return
+
+            if (this.handlePowder()) return
+
+            val entry = entries.firstOrNull { it.guiNamePattern.matches(item.name) } ?: return
+            entry.slot = this
+
+            val lore = item.getLore().takeIf { it.isNotEmpty() } ?: return
+
+            if (entry != PEAK_OF_THE_MOUNTAIN && notUnlockedPattern.matches(lore.last())) {
+                entry.activeLevel = 0
+                entry.enabled = false
+                entry.isUnlocked = false
+                return
+            }
+
+            entry.isUnlocked = true
+
+            entry.activeLevel = levelPattern.matchMatcher(lore.first()) {
+                group("level").toInt()
```

Comment:
with the previous suggestion, something like this too to save the correct level
```suggestion
                val level = group("level").toInt()
                if (group("color") == "b") level - 1 else level
```